src/lib/releaseTestPlanGenerator.ts

```diff
@@ -0,0 +1,171 @@
+import parse from "csv-parse/lib/sync";
+import { exportTestPlan } from "./testPlanFormatter";
+import {
+   findAllFilesForPattern,
+   getFilename as getFilenameFromCsv,
+   readContentFromFile,
+} from "./fileUtilities";
+import {
+   TestCase,
+   TestPlan,
+   testCaseExists,
+   findIndexForTestCase,
+   sortTestCases,
+} from "./testPlan";
+
+const RECORD_DELIMITER = ["\n", "\r\n", "\r"];
+
+/**
+ * Iterate over multiple feature test plans to extract all test cases
+ * and export them into a consolidated release test plan (csv format)
+ * @param filename
+ * @param versionToTest
+ * @param testPlanDir
+ */
+export async function generateReleaseTestPlan(
+   filename: string,
+   versionToTest: string,
+   testPlanDir: string
+): Promise<string> {
+   const files = await findTestPlanFiles(testPlanDir);
+
+   const testPlans = await Promise.all(
+      files.map(filename => loadTestPlanFromFile(filename))
+   );
+
+   // Multiple test plans might require the same test case
+   // which requires merging to consolidate their required status
+   const mergedTestCases: TestCase[] = [];
+
+   testPlans
+      .filter(testPlan => !!testPlan.testCases)
+      .map(testPlan => {
```

Comment:
This map would be better as a `.forEach` because `.map` should return something for each loop

I also think that regular `for...of` loops can be easier to read for this kind of imperative code

```js
for (testPlan of testPlans) {
  if (!!testPlan.testCases) {
    for (testCase of testPlan.testCases) {
      mergeTestCase(...);
    }
  }
}
```

Reply:
Lol it started as a for, then I changed it to a map to look more functional.  But really, code don't lie.  Back to for!